src/overview.md

```diff
@@ -0,0 +1,126 @@
+---
+layout: page
+title: Overview
+---
+
+# Overview
+
+Workbox is a collection of libraries and build tools that make it easy to
+store your website's resources locally, on your users' devices. Consider Workbox
+if you want to:
+
+* Make your site work offline.
+* Improve load performance on repeat-visits. You can use Workbox to store
+  and serve common resources locally, rather than from the network.
+
+The only difference between the two options is whether you store all or only
+some resources locally.
+
+## The technology behind Workbox
+
+Modern offline web apps are possible thanks to **service workers**. Service
+workers are background workers, written in JavaScript, that can locally save
+all of the resources needed to run your site offline.
+
+You can think of a service worker as a [proxy server][proxy] that's running in
+the background of your users' devices. Your site makes a request for a resource,
+the service worker intercepts the request, and then decides whether to serve a
+local version of the resource or to fetch a fresh version from the network.
+
+[proxy]: https://en.wikipedia.org/wiki/Proxy_server
+
+The place where Workbox stores resources locally is called the **cache**.
+The act of storing locally is called **caching**. This is because the
+underlying technology that Workbox relies on is called the [Cache API][cache].
+This is completely different than [HTTP caching][http]. The main benefit
+that Workbox provides over HTTP caching is that it can cache resources *before*
+they're requested, whereas HTTP caching can only do so after the user goes to
+a page where the resource is required.
+
+[cache]: https://developer.mozilla.org/en-US/docs/Web/API/Cache
+[http]: https://developers.google.com/web/fundamentals/performance/optimizing-content-efficiency/http-caching
+
+The logic for determining whether to serve a resource from the cache or to
+fetch a fresh version is called the **caching strategy** for that resource.
+It's common to use different caching strategies for different types of
+resources. For example:
+
+* For a logo that doesn't change often, you can probably serve it from the
+  cache.
+* For a JSON file containing a user's social media feed, you probably want
+  to display the cached content only while you fetch the latest content.
+
+## What problems Workbox solves
+
+In short, Workbox makes it easy to create optimal service worker code.
+
+Manually maintaining caches is tedious and full of pitfalls.
+The "core" Workbox libraries and tools --- `workbox-sw`, `workbox-build`,
```

Comment:
Could we make the distinction that `workbox-sw` is the runtime library, and we expect that most developers will end up using it, either directly or indirectly via the generated code.

The distinction is that only **one** of `workbox-build`, `workbox-webpack-plugin` or `workbox-cli` would need to be used—the developer would choose the one most appropriate to their build environment. The way it's written, it makes it sound like all the 4 modules do the same thing, and that a developer might end up using any/all of them.